include/libtorrent/address.hpp

```diff
@@ -58,6 +61,28 @@ namespace libtorrent {
 	using address_v6 = boost::asio::ip::address_v6;
 #endif
 #endif // SIMULATOR
+
+// the from_string member functions are deprecated starting
+// in boost 1.66.0
+#if BOOST_VERSION >= 106600 && !defined TORRENT_BUILD_SIMULATOR
+	inline address make_address(string_view str, boost::system::error_code& ec)
```

Comment:
Instead of having inline functions here, I think you could just do:

```
using boost::asio::ip::make_address;
using boost::asio::ip::make_address_v4;

```
etc. That way you would get the overloads that throw too (which I expect exist).

Reply:
one question: I thought about it, but I was asking myself if pulling the overloads that throw is a good idea. Don't  you prefer the ones that use the `error_code` to avoid  unexpected exceptions?

Reply:
I prefer exceptions for unexpected errors (and ``error_code`` for expected errors)

Reply:
OK, I will do the `using`